src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -604,59 +606,51 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
                         ty::AssocKind::Const => "associatedconstant",
                         ty::AssocKind::Type => "associatedtype",
                     };
-                    Some(if extra_fragment.is_some() {
-                        Err(ErrorKind::AnchorFailure(AnchorFailure::RustdocAnchorConflict(root_res)))
+                    return Some(if extra_fragment.is_some() {
```

Comment:
I *really* wanted to commit that this `return` makes the `else` in L613 redundant. Maybe introducing a variable could reduce any possible confusion here?

Reply:
Sure, done.